syne_tune/tuner.py

```diff
@@ -220,17 +220,24 @@ def _sleep(self):
         for callback in self.callbacks:
             callback.on_tuning_sleep(self.sleep_time)
 
+    @staticmethod
+    def _set_metadata(metadata: Dict, name: str, value):
+        if name not in metadata:
+            metadata[name] = value
+        else:
+            logger.info(f"Entry {name} already taken in metadata: {value}")
```

Comment:
I ran into this because the CLI is in fact using 'backend' in metadata. So just overwriting it here is a change that is not backward compatible.

Reply:
In the end, I'd prefer to clarify the structure and content of metadata, but until then we don't do changes that invalidate earlier results. My plotting scripts (and Aarons) use the metadata in order to do stuff.

Reply:
I did not mean to say that it is backward incompatible but that downstream functionalities (such as plotting) may easily break as they would expect a key to be something but it would have been overridden. As such, I think it makes sense to catch this potential problem early on (a logging would be a bit silent).

Would it be an option to throw a warning there and say that overriding keys will be deprecated in a newer version so that we have time to change those keys?

Reply:
OK, I do that.

Reply:
The current practice of using the class name is not great, because of the class name changes, but the concept stays the same, all sorts of result data will become hard to use in the future.

This is why I think we really need to have a clean and stable definition of this metadata, at least the relevant part of it. My practice of just dumping the CLI arguments is not good either (even though I may dump them still in the future, but into a sub-dict).

Reply:
I agree, it was more of a temporary thing. One thing that we could do is dump the string representation or the state if we have json support (easy for backend, for scheduler its not available for all we could fallback to string representation in cases that dont support it though).

Reply:
We are not indicating that overriding keys would not be allowed in the future, lets discuss offline what would be the best options to avoid potential overriding issues.